activesupport/lib/active_support/core_ext/array/access.rb

```diff
@@ -53,6 +53,19 @@ def without(*elements)
     excluding(*elements)
   end
 
+  # Returns a copy of the Array excluding the specified indiciec.
+  #
+  #   ["a", "b", "c", "d", "e", "f", "g", "h"].except_index(0, 1, -1) # => ["c", "d", "e", "f", "g"]
+  #   ["a", "b", "c", "d", "e", "f", "g", "h"].except_index(0..1, -2..-1) # => ["c", "d", "e", "f"]
+  #   ["a", "b", "c", "d", "e", "f", "g", "h"].except_index(0..1, 3, -2..-1) # => ["c", "e", "f"]
+  def except_index(*indexes)
+    to_delete = Array.new(length)
+    indexes.each do |ind|
+      ind.is_a?(Range) ? ind.each { |i| to_delete[i] = true } : to_delete[ind] = true
```

Comment:
Should we find a way to not iterate over a range once it surpasses array length? Thinking about edge cases here. 

For example, with this change:

```
class Array
  def except_index_two(*indexes)
    to_delete = Array.new(length)
    indexes.each do |ind|
      ind.is_a?(Range) ? ind.each { |i| i > length ? break : to_delete[i] = true } : to_delete[ind] = true
    end
    reject.with_index { |_, ind| to_delete[ind] }
  end
end
```

And comparison:

```
Benchmark.ips do |x|
   x.report('except_index') { %w( a b c d e f g h ).except_index((0..10000), 3, -2..-1, 99, 86, 10) }
   x.report('except_index_two') { %w( a b c d e f g h ).except_index_two((0..10000), 3, -2..-1, 99, 86, 10) }
   x.compare!
end
```
```
Warming up --------------------------------------
        except_index   130.000  i/100ms
    except_index_two    18.929k i/100ms
Calculating -------------------------------------
        except_index      1.250k (±14.2%) i/s -      6.110k in   5.048725s
    except_index_two    226.161k (± 8.5%) i/s -      1.136M in   5.073706s

Comparison:
    except_index_two:   226161.4 i/s
        except_index:     1250.4 i/s - 180.87x  slower
```

Reply:
Hey! Cool suggestion, thank you!